### ISSUE TITLE
When an Outpost is destroyed, the Rocket Men inside will walk a while without responding to attack commands

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#51]
When an Outpost is destroyed, the Rocket Men inside will walk a while without responding to attack commands.

**Comment by ExiLe-ZH** [comment]
I feel it's both an intended feature but bugged aswell.

Balance wise it would be really bad if all inf starts auto-attacking when the outpost is destroyed, this will make the unit massively more dangerous without requiring any micro (manual evac neither doesn't drop off all inf at once).

Sometimes infantry will keep walking and not respond at all, this is the bugged part I feel. 

Improving the response time would be a good.

**Comment by xezon** [comment]
Does infantry take damage after outpost popped?

**Comment by commy2** [comment]
> Does infantry take damage after outpost popped?

10% for vChina, Nuke and Tank

50% for Inf

**Comment by Jundiyy** [comment]
This is a tricky one, would be nice to fix but it changes the outcome a lot. Currently all players know you can charge at outposts as long as you have enough initla firepower, since the units inside won't attack back once the Outposts have been popped. But if fixed, it would mean you wouldn't be able to attack anymore.
In some cases I suppose this helps Chinas but maybe not for Inf. A tricky one.
P.S. I don't think it can be fixed right?

**Comment by ExiLe-ZH** [comment]
So is actually a specific outpost bug or is it related to unit spacing, the mechanic where units want to spread out?

**Comment by Stubbjax** [comment]
> 
> 
> So is actually a specific outpost bug or is it related to unit spacing, the mechanic where units want to spread out?

There is a `ScatterNearbyOnExit` field on the Outpost's `TransportContain` module, however it is set to false. I'm also curious what `GoAggressiveOnExit` does.

It seems that any infantry in a transport who is locked onto and in range of a target when the transport is destroyed will attempt to walk to the target's position while ignoring commands. This happens for any vehicle that passengers can fire out of.

**Comment by FinAminToastCrunch** [comment]
I think the issue is they the walking time of the infantry is inconsistent.

**Comment by RisingZH** [comment]
Don't like changing this one, would alter balance significantly and it seems logical that infantry wouldn't be able to fire instantly once their vehicle is destroyed.

**Comment by commy2** [comment]
Problem is not them not firing instantly, but completely ignoring any orders given.

**Comment by RisingZH** [comment]
But then I could just press stop and attack (essentially allowing them to fire instantly)

**Comment by commy2** [comment]
Sure. Kind of different meaning of "fire instantly" though.

**Comment by Madrageeeeee** [comment]
Agree not to change this too much.

would make sense to have the ‘not being able go fire’ time being the same all the time instead of different between situations?

**Comment by ExiLe-ZH** [comment]
Alternatively could set a fixed delay after which the infantry responds consistently, one that doesn't feel too fast (overpowered) or too slow (underpowered and frustrating). This needs research and experimentation.

**Comment by xezon** [comment]
> Alternatively could set a fixed delay after which the infantry responds consistently, one that doesn't feel too fast (overpowered) or too slow (underpowered and frustrating). This needs research and experimentation.

This is a reasonable suggestion. To preserve the penalty for not evacuating in time.

(It is a bit unfortunate that this report is closed as a duplicate)

**Comment by Stubbjax** [comment]
> Alternatively could set a fixed delay after which the infantry responds consistently, one that doesn't feel too fast (overpowered) or too slow (underpowered and frustrating). This needs research and experimentation.

Unresponsiveness leads to frustration. Taking control away from players is rarely the best solution to a problem, especially in a micro-focused game like Zero Hour where high player agency / low unit autonomy is a core paradigm. Such behaviour is also inconsistent with every other case of surviving occupants being force-evacuated from a destroyed container (e.g. Ambulance, Technical, Troop Crawler, Bunker, buildings).

Any imbalances are ideally addressed at their source instead of introducing or preserving bespoke and counterintuitive behaviours. If the issue is that Assault Outposts are too strong, then other options should be first considered, like standardising health (the Assault Outpost has +60 HP over the standard Listening Outpost) or counterbalances like fixing the [horde bonus bug](https://github.com/TheSuperHackers/GeneralsGamePatch/issues/810).

If we absolutely must go down the path of forced unresponsiveness, keep in mind that it needs to apply to all Listening Outpost variants, which is not ideal if the primary concern is with Infantry General. It also needs to be clearly communicated to players, such as with a disabled status and icon. This looks silly to me.

<img width="386" height="273" alt="Image" src="https://github.com/user-attachments/assets/dec163ba-1b10-4631-b815-ad8cb6c088e9" /></p>

> This is a reasonable suggestion. To preserve the penalty for not evacuating in time.

A penalty already exists - occupants take 50% damage, as is defined by the `TransportContain` module. The `ExitDelay` of 250 can also be adjusted to further encourage or discourage early evacuation.
```
Behavior = TransportContain ModuleTag_06
  DamagePercentToUnits  = 50%
End
```

**Comment by xezon** [comment]
Also fair points. Indeed the damage penalty is already convincing.

**Comment by ExiLe-ZH** [comment]
I agree unresponsiveness is a bad thing in the vast majority of cases, however some actions do have a delay, like evacing helicopters for example. As long as there's clear visual feedback I think it can be untuitive. In case of the outpost it makes sense that the infantry spreads out to take positions (to normal infantry spacing), instead of a blob of clipping infantry that can be directly ordered to fire like this. The latter is definitely way too overpowered and a no go in terms of balance.
